File: scripts/verify_score_tier_up_art.lua
package.path="./?.lua;./?/init.lua;"..package.path
local fixture=require("scripts.forest_render_fixture")
local Art=require("src.score_tier_up_art")
local function font(path,size)return{path=path,size=size,getHeight=function()return size end}end
local fonts={micro=font("assets/font-korean-regular.ttf",12),big=font("assets/font-korean-bold.ttf",28),display=font("assets/font-korean-bold.ttf",42)}
for _,t in ipairs({.08,.28,.48,.72})do
    fixture.reset()
    Art.draw({t=t,duration=.86,toTier=4},fonts,1280,720)
    assert(#fixture.commands>=3,"tier-up frame emitted too few authored layers")
    local image,text=false,0
    for _,command in ipairs(fixture.commands)do
        image=image or(command.op=="draw"and command.file=="assets/fx/score-tier-up-atlas-pixel-v1.png")
        if command.op=="text"then text=text+1 end
    end
    assert(image and text==4,"tier-up frame lost its atlas or minimal stage labels")
end
print("SCORE_TIER_UP_DRAW_OK frames=4 atlas+labels")

File: scripts/verify_score_tier_up_art.py
from pathlib import Path
from PIL import Image

root=Path(__file__).resolve().parents[1]
path=root/"assets/fx/score-tier-up-atlas-pixel-v1.png"
im=Image.open(path).convert("RGBA")
assert im.size==(9216,384),im.size
colors=set(im.get_flattened_data());assert len(colors)>=12,len(colors)
alpha={p[3] for p in colors};assert alpha<={0,255},alpha
boxes=[]
for i in range(12):
    f=im.crop((i*768,0,(i+1)*768,384));box=f.getbbox();assert box, i;boxes.append(box)
assert len(set(boxes))>=6,boxes
print("SCORE_TIER_UP_ART_OK",len(colors),"colors",len(set(boxes)),"silhouettes")
